arc/database/api.py

```
@@ -378,6 +390,126 @@ async def record_batch_outcomes(
+        """
+        try:
+            # Extract scenario ID from dict or object
+            if isinstance(scenario, dict):
+                scenario_id = scenario.get("id", f"scenario_{call_index}")
+            elif hasattr(scenario, 'id'):
+                scenario_id = scenario.id
+            else:
+                scenario_id = f"scenario_{call_index}"
+            
+            # Prepare trajectory with required fields
+            execution_time = datetime.now(timezone.utc)
```

Ruff (0.11.9)

Use `datetime.UTC` alias

Convert to `datetime.UTC` alias

(UP017)